modules/imgcodecs/test/test_grfmt.cpp

```diff
@@ -71,7 +71,7 @@ TEST_P(Imgcodecs_FileMode, regression)
 
 const string all_images[] =
 {
-#if defined(HAVE_JASPER) && defined(OPENCV_IMGCODECS_ENABLE_JASPER_TESTS)
```

Comment:
Please use:
```
#if (defined(HAVE_JASPER) && defined(OPENCV_IMGCODECS_ENABLE_JASPER_TESTS)) \
    || HAVE_OPENJPEG
```

and revert related changes in CMake file.

Reply:
You need to use `\` with multi-line directives.

Reply:
CI already told me ;-)

Reply:
done